src/AspNet.Security.OAuth.DigitalOcean/DigitalOceanAuthenticationHandler.cs

```diff
@@ -0,0 +1,117 @@
+        var tokenRequestParameters = new Dictionary<string, string?>()
+        {
+            ["client_id"] = Options.ClientId,
+            ["client_secret"] = Options.ClientSecret,
+            ["redirect_uri"] = context.RedirectUri,
+            ["code"] = context.Code,
+            ["grant_type"] = "authorization_code",
+        };
+
+        // PKCE https://tools.ietf.org/html/rfc7636#section-4.5, see BuildChallengeUrl
+        if (context.Properties.Items.TryGetValue(OAuthConstants.CodeVerifierKey, out var codeVerifier))
+        {
+            tokenRequestParameters.Add(OAuthConstants.CodeVerifierKey, codeVerifier!);
+            context.Properties.Items.Remove(OAuthConstants.CodeVerifierKey);
+        }
+
+        var address = QueryHelpers.AddQueryString(Options.TokenEndpoint, tokenRequestParameters);
+
+        using var request = new HttpRequestMessage(HttpMethod.Post, address);
+
+        using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            await Log.ExchangeCodeAsync(Logger, response, Context.RequestAborted);
+            return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+        }
+
+        using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+        var payload = JsonDocument.Parse(stream);
+        return OAuthTokenResponse.Success(payload);
+    }
+
+    protected override async Task<AuthenticationTicket> CreateTicketAsync(
+        [NotNull] ClaimsIdentity identity,
+        [NotNull] AuthenticationProperties properties,
+        [NotNull] OAuthTokenResponse tokens)
+    {
+        // Note: DigitalOcean doesn't provide a user information endpoint
+        // so we rely on the details sent back in the token request.
+        var user = tokens.Response!.RootElement.GetProperty("info");
+
+        using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+        using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+        if (!response.IsSuccessStatusCode)
+        {
+            // await Log.UserProfileErrorAsync(Logger, response, Context.RequestAborted);
+            throw new HttpRequestException("An error occurred while retrieving the user profile.");
+        }
+
+        using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync(Context.RequestAborted));
+
+        var account = payload.RootElement.GetProperty("account");
+
+        identity.AddClaim(new Claim("email_verified", account.GetProperty("email_verified").ToString().ToLowerInvariant()));
```

Comment:
What is the type of the `email_verified` property? If it's a string, you can use `GetString("email_verified")`. If it's a boolean, use `GetBoolean()` before calling `ToString()`.

Reply:
`email_verified` is if the user's email address has been verified or not. It's really the only other "user" info.

The rest of the account info is related to account limits/billing/etc

Reply:
My question was more about the type. `ToString()` just forces whatever it is into a string.

For example if it were an object, then it'd return `{"orly?":"yas"}`. If it's a boolean `true`, using `GetBoolean()` on it would throw if value wasn't a boolean.

Where as if it was a string of `"true"`, then `GetString("email_verified")` can be used as it's shorthand for `GetProperty("email_verified").GetString()`.

Reply:
Makes sense.  It's a `bool`, should I still `ToLowerInvariant()`?

Reply:
> identity.AddClaim(new Claim(DigitalOceanAuthenticationConstants.Claims.EmailVerified, account.GetProperty("email_verified").GetBoolean().ToString().ToLowerInvariant()));

Reply:
I think lowering it should be fine 👍